src/asyncify/asyncify_prompt_impl.c

```diff
@@ -0,0 +1,280 @@
+// An asyncify implementation of the basic fiber interface.
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <assert.h>
+
+#include <wasi-io.h>
+#include <fiber/prompt.h>
+#define import(NAME) __attribute__((import_module("asyncify"),import_name(NAME)))
+
+
+/** Asyncify imports **/
+// The following functions are asyncify primitives:
+// * asyncify_start_unwind(iptr): initiates a continuation
+//   capture. The argument `iptr` is a pointer to an asyncify stack.
+// * asyncify_stop_unwind(): delimits a capture continuations.
+// * asyncify_start_rewind(iptr): initiates a continuation
+//   reinstatement. The argument `iptr` is a pointer to an asyncify
+//   stack.
+// * asyncify_stop_rewind(): delimits the extent of a continuation
+//  reinstatement.
+extern
+import("start_unwind")
+void asyncify_start_unwind(void*);
+
+extern
+import("stop_unwind")
+void asyncify_stop_unwind(void);
+
+extern
+import("start_rewind")
+void asyncify_start_rewind(void*);
+
+extern
+import("stop_rewind")
+void asyncify_stop_rewind(void);
+
+volatile uint32_t asyncify_state = 0;
+
+// The default stack size is 2MB.
+static const size_t default_stack_size = ASYNCIFY_DEFAULT_STACK_SIZE;
+
+// We track the currently active fiber via this global variable.
+static volatile fiber_t active_fiber = NULL;
+
+static volatile yield_result_t fiber_args = {0, NULL};
+
+// Prompt generator
+static uint32_t next_prompt = 0;
+
+// Fiber states:
+// * ACTIVE: the fiber is actively executing.
+// * YIELDING: the fiber is suspended.
+// * DONE: the fiber is finished (i.e. run to completion).
+typedef enum { ACTIVE, YIELDING, DONE, FORWARDING, YIELD_FORWARDING } fiber_state_t;
```

Comment:
i think `YIELD_FORWARDING` might not be used anymore

Reply:
Correct! Thanks.